### PR TITLE
MJ Consumable

### PR DESCRIPTION
**Pull request by RegisCoaxans**
## Description
Added a new consumable that only Magikarp species can use. It acts like stronger Rare Candies.
It is a Rare Hold Item on all Magikarp (except base) on routes for now. More obtaining methods in a later PR.
The images are from the game, on Bulbapedia.

## Motivation and Context
Part of a new iteration on Magikarp Jump subregion.

## How Has This Been Tested?
I gave myself a hell of a lot of that consumable and used them on my Karp.

## Types of changes
- New feature

### CODE REVIEW

**Comment by osherwott** [comment]
Any particular reason there's 3 of them? I'm hesitant on using Lava Cookies and Malasadas for Karp Jump, because we might want to use them for other things. Might be able to substitute on of them with Magikarp Jump's "Support Candy", however.
image
https://archives.bulbagarden.net/wiki/File:Magikarp_Jump_Support_Candy.png

**Comment by RegisCoaxans** [comment]
All three are from the MJ game. Cookie and Malasada have effects we can not use in PokéClicker so I thought they would fit. I can switch to other items from the game, however the candy is boring...

**Comment by RegisCoaxans** [comment]
All Magikarp Jump consumables are either berries or regional specialities, though...

**Comment by osherwott** [comment]
For items that could be used as consumables (besides the food we already mentioned), there are:
- Training Sodas. These look a bit like Soda Pops, but they could work
image
- Power Ups. Personally, I don't like these
image
- Skill Restores. Weird, but preferable over Power Ups
image
https://bulbapedia.bulbagarden.net/wiki/Pok%C3%A9mon:_Magikarp_Jump#One-time-use_items

If none of these are desirable, then we can still use the regional snacks, no prob 👍

**Comment by RegisCoaxans** [comment]
I like none. Having them all being pastries looks neat... Are there no other specialities you could book for your idea ? Like, I can use Shalour Sable and leave Lumiose Galette to you or the opposite, and I just need one more pastry from these available in the MJ.

**Comment by wormania** [comment]
With the 3 items all being effectively the same thing but with number differences, what about something like recoloured "flavoured" biscuits for the alternates, e.g. a strawberry biscuit:
Strawberry_Magikarp_Biscuit

**Comment by RegisCoaxans** [comment]
I think we will just go with different pastries from the MJ game. Thanks for the idea anyway, I liked it.

File: src/scripts/pokemons/PokemonFactory.ts [hunk 1]
```diff
@@ -316,6 +316,8 @@ class PokemonFactory {
             case 'White_mane_hair':
                 chance = GameConstants.MANE_ITEM_CHANCE;
                 break;
+            case 'Magikarp_Biscuit':
+                chance = GameConstants.HELD_MAGIKARP_BISCUIT;
```
[the remark this answers is not in the document]
Comment (RegisCoaxans) [reply src/scripts/pokemons/PokemonFactory.ts:309]: Added. Also added one further down after Christmas Present case.

**Pull request merged**